shells/android/javatests/arcs/storage/service/BindingContextTest.kt

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.storage.service
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.crdt.CrdtCount
+import arcs.crdt.internal.VersionMap
+import arcs.crdt.parcelables.ParcelableCrdtType
+import arcs.data.CountType
+import arcs.storage.ExistenceCriteria
+import arcs.storage.ProxyMessage
+import arcs.storage.Store
+import arcs.storage.StoreOptions
+import arcs.storage.driver.RamDisk
+import arcs.storage.driver.RamDiskDriverProvider
+import arcs.storage.driver.RamDiskStorageKey
+import arcs.storage.parcelables.toParcelable
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.yield
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/** Tests for [BindingContext]. */
+@RunWith(AndroidJUnit4::class)
+@UseExperimental(ExperimentalCoroutinesApi::class, FlowPreview::class)
+class BindingContextTest {
+    private lateinit var store: Store<CrdtCount.Data, CrdtCount.Operation, Int>
+
+    @Before
+    fun setup() {
+        RamDiskDriverProvider()
+        RamDisk.clear()
+        store = Store(
+            StoreOptions(
+                RamDiskStorageKey("myCount"),
+                ExistenceCriteria.ShouldCreate,
+                CountType()
+            )
+        )
+    }
+
+    @Test
+    fun getLocalData_fetchesLocalData() = runBlocking {
```

Comment:
I've seen you use `runBlockingTest` elsewhere. Should you use that here? What's the difference?

Reply:
I had to use `runBlocking` here, and in a few other places, because runBlockingTest has an outstanding bug in certain situations - unfortunately. Though I did verify that there shouldn't be any outstanding jobs.